sklearn/utils/estimator_checks.py

```diff
@@ -2345,3 +2347,63 @@ def check_outliers_fit_predict(name, estimator_orig):
         for contamination in [-0.5, 2.3]:
             estimator.set_params(contamination=contamination)
             assert_raises(ValueError, estimator.fit_predict, X)
+
+
+def check_fit_idempotent(name, estimator_orig):
+    # Check that est.fit(X) is the same as est.fit(X).fit(X). Ideally we would
+    # check that the estimated parameters during training (e.g. coefs_) are
+    # the same, but having a universal comparison function for those
+    # attributes is difficult and full of edge cases. So instead we check that
+    # predict(), predict_proba(), decision_function() and transform() return
+    # the same results.
+
+    est = clone(estimator_orig)
+
+    np.random.seed(0)
+    X = np.random.normal(loc=100, size=(100, 2))
+    if is_regressor(estimator_orig):
+        y = np.random.normal(size=100)
+    else:
+        y = np.random.randint(low=0, high=2, size=100)
+    if est.__class__.__name__.startswith('MultiTask'):
+        y = np.column_stack([y, y])
+
+    X_train, X_test, y_train, _ = train_test_split(X, y)
+    # some estimators expect a square matrix
+    X_train = pairwise_estimator_convert_X(X_train, est)
+    X_test = pairwise_estimator_convert_X(X_train, est)
+
+    if 'random_state' in est.get_params().keys():
+        est.set_params(random_state=0)
+    if 'warm_start' in est.get_params().keys():
+        est.set_params(warm_start=False)
+
+    est.fit(X_train, y_train)
+
+    if hasattr(est, 'predict'):
+        pred_1 = est.predict(X_test)
+    if hasattr(est, 'predict_proba'):
+        pred_proba_1 = est.predict_proba(X_test)
```

Comment:
Note that we don't need to test predict if predict_proba is tested.  This is the kind of rationalisation and optimisation that writing a generic function would benefit from where here it would look ugly.